app/workers/queue_listener.rb

```diff
@@ -43,11 +43,17 @@ def update_share(params)
   end
 
   def petition_id(params)
-    params.fetch(:petition_uri, '').match(/(\d+)\/$/)[1]
+    extract_id(params[:petition_uri])
   end
 
   def donation_id(params)
-    params.fetch(:donation_uri, '').match(/(\d+)\/$/)[1]
+    extract_id(params[:donation_uri])
+  end
+
+  def extract_id(uri)
+    id = (uri || '').match(/(\d+)\/$/){|m| m[1] if m }
```

Comment:
I'm trying to extract the ID from the URI. The block allows me to check if there was a match before accessing it.